code/modules/mob/living/status_procs.dm

```diff
@@ -241,6 +241,66 @@
 			P = apply_status_effect(STATUS_EFFECT_PARALYZED, amount)
 		return P
 
+///////////////////////////////// STARTLED //////////////////////////////////
+
+/mob/living/proc/IsStartled() //If we're Startled
+	return has_status_effect(STATUS_EFFECT_STARTLED)
+
+/mob/living/proc/AmountStartled() //How many deciseconds remain in our Startled status effect
+	var/datum/status_effect/incapacitating/startled/I = IsStartled()
+	if(I)
+		return I.duration - world.time
+	return 0
+
+/mob/living/proc/Startle(amount, ignore_canstun = FALSE) //Can't go below remaining duration
+	if(SEND_SIGNAL(src, COMSIG_LIVING_STATUS_STARTLE, amount, ignore_canstun) & COMPONENT_NO_STUN)
+		return
+	if(status_flags & GODMODE)
+		return
+	if(((status_flags & CANKNOCKDOWN) && !HAS_TRAIT(src, TRAIT_STUNIMMUNE)) || ignore_canstun)
+		if(absorb_stun(amount, ignore_canstun))
+			return
+		var/datum/status_effect/incapacitating/startled/I = IsStartled()
+		if(I)
+			I.duration = max(world.time + amount, I.duration)
+		else if(amount > 0)
+			I = apply_status_effect(STATUS_EFFECT_STARTLED, amount)
+		return I
+
+/mob/living/proc/SetStartled(amount, ignore_canstun = FALSE) //Sets remaining duration
+	if(SEND_SIGNAL(src, COMSIG_LIVING_STATUS_STARTLE, amount, ignore_canstun) & COMPONENT_NO_STUN)
+		return
+	if(status_flags & GODMODE)
+		return
+	if(((status_flags & CANKNOCKDOWN) && !HAS_TRAIT(src, TRAIT_STUNIMMUNE)) || ignore_canstun)
+		var/datum/status_effect/incapacitating/startled/I = IsStartled()
+		if(amount <= 0)
+			if(I)
+				qdel(I)
+		else
+			if(absorb_stun(amount, ignore_canstun))
+				return
+			if(I)
+				I.duration = world.time + amount
+			else
+				I = apply_status_effect(STATUS_EFFECT_STARTLED, amount)
+		return I
+
+/mob/living/proc/AdjustStartled(amount, ignore_canstun = FALSE) //Adds to remaining duration
+	if(SEND_SIGNAL(src, COMSIG_LIVING_STATUS_STARTLE, amount, ignore_canstun) & COMPONENT_NO_STUN)
+		return
+	if(status_flags & GODMODE)
+		return
+	if(((status_flags & CANKNOCKDOWN) && !HAS_TRAIT(src, TRAIT_STUNIMMUNE)) || ignore_canstun)
+		if(absorb_stun(amount, ignore_canstun))
+			return
+		var/datum/status_effect/incapacitating/startled/I = IsStartled()
```

Comment:
https://github.com/tgstation/tgstation/blob/master/.github/CONTRIBUTING.md#use-descriptive-and-obvious-names